src/ibek/pattern_cmds/sources.py

```
@@ -0,0 +1,153 @@
+def _local_path(uri: str) -> Path:
+    return Path(uri[len("file://") :] if uri.startswith("file://") else uri)
+
+
+def _clone_pattern(uri: str, name: str, version: str | None, dest: Path) -> Path:
+    """Clone ``uri`` at ``version`` and return the ``name`` pattern folder."""
+    clone_dir = dest / "clone"
+    cmd = ["git", "clone", "--depth", "1", "--quiet"]
+    if version:
+        cmd += ["--branch", version]
+    cmd += [uri, str(clone_dir)]
+    result = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)